src/views/Search.vue

```diff
@@ -69,35 +64,26 @@ export default Vue.extend({
       set(query) { this.$store.commit("searchQuery", query) }
     },
     index: {
-      get() { return this.sindex },
-      set(value) { this.sindex = Math.min(Math.max(value, 1), 10) }
+      get() { return this.$store.state.index },
+      set(value) {
+        this.$store.commit("searchIndex", Math.min(Math.max(value, 1), 10))
+        this.search()
+      }
     },
-    pages: {
-      get() { return this.spages },
-      set(value) { this.spages = Math.min(Math.max(value, 1), 3) }
-    }
   },
   mounted() {
     this.reader = this.$refs.reader
     this.search()
   },
-  watch: {
-    "$store.state.query"() {
-        this.search()
-    },
-    pages() {
-      this.search()
-    },
-    index() {
-      this.search()
-    }
-  },
   methods: {
-    search() {
+    async search() {
       // avoid stacking requests
       if (!this.$store.state.loading) {
```

Comment:
using loading state to avoid race condition is not the best pattern, race conditions can still happen and user may want to do an other search without waiting for the precedent to finish